lib/report_parser.ex

```diff
@@ -94,8 +103,17 @@ defmodule RecordingConverter.ReportParser do
   defp calculate_track_duration(track) do
     clock_rate_ms = div(track["clock_rate"], 1_000)
 
-    difference_in_milliseconds =
-      div(track["end_timestamp"] - track["start_timestamp"], clock_rate_ms)
+    end_timestamp = track["end_timestamp"]
+    start_timestamp = track["start_timestamp"]
+
+    timestamp_difference =
+      if end_timestamp < start_timestamp do
+        end_timestamp + 2 ** 32 - 1 - start_timestamp
```

Comment:
Module attribute @max_value = 2 ** 32 - 1
```suggestion
       @max_value = 2 ** 32 - 1
        end_timestamp +  @max_value - start_timestamp
```